src/analysis/properly_documented_public_functions.d

```diff
@@ -0,0 +1,202 @@
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.properly_documented_public_functions;
+
+import dparse.lexer;
+import dparse.ast;
+import analysis.base : BaseAnalyzer;
+
+import std.stdio;
+
+/**
+ * Requires each public function to contain the following ddoc sections
+	- PARAMS: (if it has at least one parameter)
+	- RETURNS: (except if it's void)
+ */
+class ProperlyDocumentedPublicFunctions : BaseAnalyzer
+{
+	enum string MISSING_PARAMS_KEY = "dscanner.style.doc_missing_params";
+	enum string MISSING_PARAMS_MESSAGE = "A public function needs to contain a `Params` section";
+
+	enum string MISSING_RETURNS_KEY = "dscanner.style.doc_missing_returns";
+	enum string MISSING_RETURNS_MESSAGE = "A public function needs to contain a `Returns` section";
+
+	///
+	this(string fileName, bool skipTests = false)
+	{
+		super(fileName, null, skipTests);
+	}
+
+	override void visit(const Module mod)
+	{
+		mod.accept(this);
+	}
+
+	override void visit(const Declaration decl)
+	{
+		import std.algorithm.searching : any;
+		import std.algorithm.iteration : map;
+
+		// skip private symbols
+		enum tokPrivate = tok!"private", tokProtected = tok!"protected", tokPackage = tok!"package";
```

Comment:
No, this was just a "hack" to only look at publicly exposed functions. I added another check to look out for `private` etc. labels.